sdk/storage/Azure.Storage.Blobs/src/Models/GetBlobContainersAsyncCollection.cs

```diff
@@ -26,19 +27,18 @@ internal class GetBlobContainersAsyncCollection : StorageCollectionEnumerator<Bl
         public override async ValueTask<Page<BlobContainerItem>> GetNextPageAsync(
             string continuationToken,
             int? pageSizeHint,
-            bool isAsync,
+            bool async,
             CancellationToken cancellationToken)
         {
-            Task<Response<BlobContainersSegment>> task = _client.GetBlobContainersInternal(
-                continuationToken,
-                _traits,
-                _prefix,
-                pageSizeHint,
-                isAsync,
-                cancellationToken);
-            Response<BlobContainersSegment> response = isAsync ?
-                await task.ConfigureAwait(false) :
-                task.EnsureCompleted();
+
+            Response<BlobContainersSegment> response = await _client.GetBlobContainersInternal(
+                    continuationToken,
+                    _traits,
+                    _prefix,
+                    pageSizeHint,
+                    async,
+                    cancellationToken).ConfigureAwait(false);
```

Comment:
This used to call EnsureCompleted() and do the debug verification but now it doesn't.  Is there a reason we're choosing not to?  This question applies to all the Storage enumerators that have a similar change.

Reply:
It is safe for `async` parameter to pass it from calling method assuming that somewhere in the end it will be checked (for that analyzer validates all methods with `async` parameter). In this case, `GetBlobContainersInternal` passes it to `ListBlobContainersSegmentAsync` which is executed synchronously when `async == false`. So `await` operation want do anything and `ListBlobContainersSegmentAsync` will return completed task. As a result, there will be no async action in `GetBlobContainersInternal` and it will also return completed task. This sample will be executed 100% synchronously if `async == false`:
```cs
private async Task<int> M1Async(bool async) {
    var t = M2Async(async);
    Debug.Assert(t.IsCompleted);
    var x = await t.ConfigureAwait(false);
    return x + 1;
}

private async Task<int> M2Async(bool async) {
    var t = M3Async(async);
    Debug.Assert(t.IsCompleted);
    var x = await t.ConfigureAwait(false);
    return x + 2;
}

private async Task<int> M3Async(bool async) {
    if (async) {
        await Task.Delay(10).ConfigureAwait(false);
    } else {
        Thread.Sleep(10);
    }
    return 3;
}
```

Reply:
Doesn't that make it harder to debug though?  I get that we'll still catch it at the edge of the library, but by not having the verification at every level we'll just know that somebody blocked somewhere in the event of failure and have to spend time hunting down the cause.

Reply:
In this particular case, I don't think there is any benefit in debugging, cause these methods are basically wrappers. And it's a job of analyzer to check that `async` value is passed correctly.